featurizers/fingerprints.py

```diff
@@ -336,46 +339,48 @@ def __init__(
         self.pool_multiplier = pool_multiplier
         self.rmsd_cutoff = rmsd_cutoff
         self.max_energy_diff = max_energy_diff
-        self.force_field = force_field
+        self.forcefield = forcefield
         self.get_values = get_values
         self.is_folded = is_folded
         self.fold_bits = fold_bits
         self.standardise = standardise
-        self.random_state = random_state
+        self.seed = seed
 
     def _calculate_fingerprint(
         self, X: Union[pd.DataFrame, np.ndarray]
-    ) -> np.ndarray:
+    ) -> Union[np.ndarray, spsparse.csr_array]:
         from e3fp.conformer.generator import ConformerGenerator
-        from e3fp.conformer.util import mol_from_smiles
         from e3fp.pipeline import fprints_from_mol
 
-        X = self._validate_input(X)
-
         conf_gen = ConformerGenerator(
             first=self.first,
             num_conf=self.num_conf,
             pool_multiplier=self.pool_multiplier,
             rmsd_cutoff=self.rmsd_cutoff,
             max_energy_diff=self.max_energy_diff,
-            forcefield=self.force_field,
+            forcefield=self.forcefield,
             get_values=self.get_values,
-            seed=self.random_state,
+            seed=self.seed,
         )
 
         result = []
 
         for x in X:
-            input_mol = mol_from_smiles(
-                smiles=MolToSmiles(x),
-                name=MolToSmiles(x),
-                standardise=self.standardise,
-            )
+            if isinstance(x, Mol):
+                smiles = MolToSmiles(x)
+                mol = x
+            else:
+                smiles = x
+                mol = MolFromSmiles(x)
```

Comment:
According to typing, X is `Union[pd.DataFrame, np.ndarray]`. This assumes that it is a list of either strings or Mol objects. I think that we assume the former in all cases, i.e. we work more like `CountVectorizer` than `PolynomialFeatures` transformer. Right? So the typing should be changed here.

Reply:
Current implementation assumes both smiles strings and mol objects. We can process smiles and turn them into molecules with use of _validate_input() from the base class. This allows us to not only compute the fingerprints concurrently but also the transformation from smiles to fingerprints, as it turned out to be slow and could potentially become a bottleneck for larger datasets. I think we decided to do that after finding out that initial benchmark scores didn't result in a great time improvement

Reply:
We could create a new SmilesToMol transformer but the time effort for that would most likely be greater than we can afford before our deadlines. I'll be happy to implement it as future work